modules/navier_stokes/src/fvkernels/PINSFVTKEDSourceSink.C

```diff
@@ -0,0 +1,194 @@
+
+  //symmetric_strain_tensor_norm = std::sqrt(symmetric_strain_tensor_norm + offset);
+  symmetric_strain_tensor_norm = 2.0 * symmetric_strain_tensor_norm + offset;
+
+  constexpr Real protection_k = 1e-15;
+
+  ADReal production, destruction;
+  auto production_k = _mu_t(makeElemArg(_current_elem)) * symmetric_strain_tensor_norm;
+  if (_linearized_model)
+  {
+    production = _C1_eps(makeElemArg(_current_elem))
+                      * production_k
+                      * _linear_variable(makeElemArg(_current_elem));
+
+    destruction = _C2_eps(makeElemArg(_current_elem))
+                      * _rho(makeElemArg(_current_elem))
+                      * _var(makeElemArg(_current_elem))
+                      * _linear_variable(makeElemArg(_current_elem));
+    
+    // _console << "Epsilon: " << _var(makeElemArg(_current_elem)) << std::endl;
+                      
+    // _console << "Linearized k epsilon model" << std::endl;
+  }
+  else
+  {
+    production = _C1_eps(makeElemArg(_current_elem))
+                      * production_k
+                      * _var(makeElemArg(_current_elem))
+                      / (_k(makeElemArg(_current_elem)) + protection_k);
+
+    destruction = _C2_eps(makeElemArg(_current_elem))
+                      * _rho(makeElemArg(_current_elem))
+                      * Utility::pow<2>(_var(makeElemArg(_current_elem)))
+                      / (_k(makeElemArg(_current_elem)) + protection_k);
+  }
+
+  // _console << "Coord: " << _current_elem->vertex_average() << std::endl;
+  // _console << "Production eps: " << production << std::endl;
+  // _console << "Destruction eps: " << destruction << std::endl;
+
+  // _console << "Production epsilon: " << production << std::endl;
+  // _console << "Dissipation epsilon: " << destruction << std::endl;
+
+  if (_realizable_constraint)
+  {
+    // Realizable dissipation constraints
+    production = (production > 0) ? production : 0.0;
+    destruction = (destruction > 0) ? destruction : 0.0;
+    destruction = (destruction < production) ? destruction : production;
+    
+  }
+
+  residual += production - destruction;
+
+  // Mixing length realizable source constraints
+  // if (_realizable_constraint)
+  // {
+  //   //auto bottom_constraint = 0.09*std::pow(_k(makeElemArg(_current_elem)), 1.5) / _max_mixing_length / 10.0; // Must include hard value for C_mu
+  //   //auto top_constraint = std::pow(_k(makeElemArg(_current_elem)), 1.5) / _max_mixing_length * 10.0;
+  //   //residual = (residual > top_constraint) ? residual : top_constraint;
+  //   //residual = (residual < bottom_constraint) ? residual : bottom_constraint;
+  //   if ( (residual < 0) && (std::abs(residual) > 0.5*_var(makeElemArg(_current_elem))) )
+  //     residual *= 0.5*_var(makeElemArg(_current_elem))  / residual;
+  //   Real scaled_argument = 1e-9/std::abs(residual.value() + 1e-12);
+  //   auto exp_argument = std::min(scaled_argument, 20.0);
+  //   residual *= std::exp(- exp_argument);
+  // }
```

Comment:
I'll stop adding `remove unused code` comments because my browser is getting slower